.github/workflows/deploy_pr_preview.yml

```diff
@@ -0,0 +1,78 @@
+---
+name: Preview pull requests
+# Setup and build files with hugo
+# https://github.com/peaceiris/actions-hugo
+# Deploy to GitHub pages
+# https://github.com/peaceiris/actions-gh-pages
+
+on:
+  workflow_run:
+    workflows: ["Build pull request preview"]
+    types: [completed]
+
+jobs:
+  deploy:
+    # Only run if PR preview build was successful
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    name: Deploy pull request preview
+    runs-on: ubuntu-18.04
+    steps:
+      - name: "Download artifact"
+        uses: actions/github-script@v3.1.0
+        with:
+          script: |
+            var artifacts = await github.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: ${{github.event.workflow_run.id }},
+            });
+            var matchArtifact = artifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "pr"
+            })[0];
+            var download = await github.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            var fs = require('fs');
+            fs.writeFileSync('${{github.workspace}}/pr.zip', Buffer.from(download.data));
+
+      - name: Extract pull request preview
+        run: unzip pr.zip
+
+      - name: Deploy Pull Request preview
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          # Use GITHUB_TOKEN allowing to write to local repository
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          # XXX add a way to clean deployed PR having been closed or merged
```

Comment:
I guess this comment can be removed now

Reply:
yep, you are right. Let's see later if it works.